### PR TITLE
Ruby 2.5+ does not need sysrandom

### LINKED ISSUE #13: sysrandom
This gem depends on [sysrandom](https://github.com/cryptosphere/sysrandom) gem, but it is archived already.

Because Ruby 2.5+ [tries to use /dev/urandom before openssl](https://bugs.ruby-lang.org/projects/ruby-trunk/repository/trunk/revisions/57384), we can rely on it and remove dependency to sysrandom if we drop support Ruby < 2.5. What do you think?

### PR DESCRIPTION
**Pull request by sakuro**
Ulid depends on _archived_ [sysrandom](https://github.com/cryptosphere/sysrandom) gem.
I think this is becase Ruby's SecureRandom [did not try /dev/urandom first](https://bugs.ruby-lang.org/issues/9569).  But this has been changed in Ruby 2.5. Ruby 2.5+'s SecureRandom does [try /dev/urandom first](https://bugs.ruby-lang.org/projects/ruby-trunk/repository/trunk/revisions/57384).
So in Ruby 2.5+, ulid does not need to depend on sysrandom.

### CODE REVIEW

**Comment by choosen** [comment]
Sysrandom is replacing existing ruby std lib constant.. 
so we don't require this gem at all. 
Developers can require this gem by their own (just mention that in readme)

from sysrandom code:
```
Object.send(:remove_const, :SecureRandom)
SecureRandom = Sysrandom
```

File: ulid.gemspec [hunk 1]
```diff
@@ -17,5 +17,5 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'sysrandom', '>= 1.0.0', '< 2.0'
+  spec.required_ruby_version = '>= 2.5.0'
```
Comment (choosen) [inline ulid.gemspec]: SecureRandom exists even in https://www.rubydoc.info/stdlib/securerandom/1.8.7/SecureRandom

so we don't need to specify version at all

Comment (sakuro) [reply ulid.gemspec]: Correct me if I'm wrong but my understanding about the reason of adopting sysrandom gem in ulid is that [SecureRandom does not use /dev/urandom first](https://bugs.ruby-lang.org/issues/9569) before 2.5. In 2.5+ [it does](https://bugs.ruby-lang.org/projects/ruby-trunk/repository/trunk/revisions/57384).

Comment (choosen) [reply ulid.gemspec]: so we can leave this dependency with constrain:
`spec.add_dependency 'sysrandom', '>= 1.0.0', '< 2.0' if RUBY_VERSION < "2.5"`

Comment (choosen) [reply ulid.gemspec]: hmm I am not sure if we can do it in gemspec ; )

Comment (sakuro) [reply ulid.gemspec]: with my commit 09e7737, now sysrandom is required if necessary.

Gemfile

```rb
# frozen_string_literal: true

source "https://rubygems.org"

git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }

# gem "rails"

gem 'ulid', github: 'sakuro/ulid', branch: 'remove-sysrandom'
```

Gemfile.lock bundled with Ruby 2.4.3 (before 2.5.0)

```
GIT
  remote: https://github.com/sakuro/ulid
  revision: 09e7737febfbefa20dd78acbf655674fb9473d7c
  branch: remove-sysrandom
  specs:
    ulid (1.0.0)
      sysrandom (>= 1.0.0, < 2.0)

GEM
  remote: https://rubygems.org/
  specs:
    sysrandom (1.0.5)

PLATFORMS
  ruby

DEPENDENCIES
  ulid!

BUNDLED WITH
   2.0.1
```

Gemfile.lock bundled with Ruby 2.6.1 (latest)

```
GIT
  remote: https://github.com/sakuro/ulid
  revision: 09e7737febfbefa20dd78acbf655674fb9473d7c
  branch: remove-sysrandom
  specs:
    ulid (1.0.0)

GEM
  remote: https://rubygems.org/
  specs:

PLATFORMS
  ruby

DEPENDENCIES
  ulid!

BUNDLED WITH
   2.0.1
```

File: ulid.gemspec [hunk 2]
```diff
@@ -17,5 +17,5 @@ Gem::Specification.new do |spec|
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ['lib']
 
-  spec.add_dependency 'sysrandom', '>= 1.0.0', '< 2.0'
+  spec.add_dependency 'sysrandom', '>= 1.0.0', '< 2.0' unless RUBY_VERSION >= '2.5'
```
Comment (choosen) [inline ulid.gemspec]: what about omitting the negation ?
`spec.add_dependency 'sysrandom', '>= 1.0.0', '< 2.0' if RUBY_VERSION < '2.5'`

File: ulid.gemspec (hunk 1 above)
Comment (choosen) [reply ulid.gemspec]: Looks good to me 👍 

(you can update PR title + description)

Comment (sakuro) [reply ulid.gemspec]: Updated title and desc.

File: ulid.gemspec (hunk 2 above)
Comment (sakuro) [reply ulid.gemspec]: Right. It is much straightforward, i'll fix it.

Comment (sakuro) [reply ulid.gemspec]: done in 9162f3a

**Review by choosen — APPROVED** [review]
LGTM 👍 
@rafaelsales  can you look at this PR ?

**Comment by rafaelsales** [comment]
Thanks guys!

**Comment by rafaelsales** [comment]
I just release v1.1.0 with these changes.

**Pull request merged**